src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -0,0 +1,289 @@
+                    this._formPoster.postForm(error.body.three_ds_result.acs_url, {
+                        PaReq: error.body.three_ds_result.payer_auth_request,
+                        TermUrl: error.body.three_ds_result.callback_url,
+                        MD: error.body.three_ds_result.merchant_data,
+                    });
+                });
+            });
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        if (this._adyenOldComponent) {
+            this._adyenOldComponent.unmount();
+            this._adyenOldComponent = undefined;
+        } else if (this._adyenComponent) {
+            this._adyenComponent.unmount();
+            this._adyenComponent = undefined;
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _getAdyenV2PaymentInitializeOptions(): AdyenV2PaymentInitializeOptions {
+        if (!this._adyenv2) {
+            throw new InvalidArgumentError(' "options.adyenv2" argument was not provided during initialization.');
+        }
+
+        return this._adyenv2;
+    }
+
+    private _getLocale(): string {
+        const state = this._store.getState();
+        const storeConfig = state.config.getStoreConfig();
+
+        if (!storeConfig) {
+            return 'en_US';
+        }
+
+        return storeConfig.storeProfile.storeLanguage;
+    }
+
+    private _getStateContainer(): string {
+        if (!this._stateContainer) {
+            return '{}';
+        }
+
+        return this._stateContainer;
+    }
+
+    private _getThreeDS2ChallengeWidgetSize(): string {
+        const { threeDS2ChallengeWidgetSize } = this._getAdyenV2PaymentInitializeOptions();
+
+        if (!threeDS2ChallengeWidgetSize) {
+            return '01';
+        }
+
+        return threeDS2ChallengeWidgetSize;
+    }
+
+    private _handle3DS2Challenge(resultObject: ThreeDS2Result, paymentMethodId: string): Promise<Payment> {
+
+        return new Promise((resolve, reject) => {
+            if (!this._adyenCheckout) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            const threeDS2Component = this._adyenCheckout
+                .create(ThreeDS2ComponentType.ThreeDS2Challenge, {
+                    challengeToken: resultObject.token,
+                    onComplete: (challengeData: ThreeDS2OnComplete) => {
+                        const challengePaymentPayload = {
+                            ...challengeData.data,
+                            paymentData: resultObject.payment_data,
+                        };
+
+                        const paymentPayload = {
+                            methodId: paymentMethodId,
+                            paymentData: {
+                                nonce: JSON.stringify(challengePaymentPayload, null, 2),
```

Comment:
If you don't need to add additional whitespace to the JSON output, I prefer you leaving out the 2nd and 3rd argument.